src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -1063,6 +1075,61 @@ public static String slurp(final InputStream is, final Charset charSet) {
         return output;
     }
 
+    /**
+     * Go through the files provided and if they have one of the provided file extensions pass the file to the output
+     * otherwise assume that file is a list of filenames and unfold it into the output (recursively).
+     */
+    public static List<Path> unrollPaths(final Collection<Path> inputs, final String... extensions) {
```

Comment:
@yfarjoun Is there a reason we can't replace the file version with a call to the new path code?

Reply:
well, it returns a List<Path> instead of a List<File>...

Reply:
You could do the conversion on the way in and out.  It's slightly less efficient, but probably worth it to avoid a bunch of lines of code.

Reply:
fair enough.